Codechange: merge guiflags and flags in settings .ini files

TrueBrain opened the pull request:
## Motivation / Problem

While working on SaveLoad I ran into the issue that it has two flags which have nothing to do with SaveLoad. Additionally, the Settings were settings SaveLoad settings directly, which gave for some confusing ini file .. `flags` vs `guiflags`.


## Description

Simplify the situation by only having `flags`.

This now means that `SF_NOT_IN_CONFIG`, `SF_NOT_IN_SAVE` and `SF_NO_NETWORK_SYNC` remain within the Setting, and don't pollute SaveLoad for no good reason.


## Limitations

<!--
Describe here
* Is the problem solved in all scenarios?
* Is this feature complete? Are there things that could be added in the future?
* Are there things that are intentionally left out?
* Do you know of a bug or corner case that does not work?
-->


## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

rubidium42 commented on src/saveload/saveload.h:
```diff
@@ -483,13 +483,10 @@ enum VarTypes {
+	SLF_NO_NETWORK_SYNC = 1 <<  8, ///< do not synchronize over network
+	SLF_ALLOW_CONTROL   = 1 <<  9, ///< allow control codes in the strings
+	SLF_ALLOW_NEWLINE   = 1 << 10, ///< allow new lines in the strings
+	SLF_HEX             = 1 << 11, ///< print numbers as hex in the config file (only useful for unsigned)
```
Practically this is also for (formatting of) the config file, so shouldn't that belong there too?

rubidium42 replied:
Missing capitalisation and periods?

rubidium42 commented on src/table/settings/gameopt_settings.ini:
```diff
@@ -14,7 +14,7 @@
 
 [pre-amble]
-static const uint GAME_DIFFICULTY_NUM = 18;
+static const uint GAME_DIFFICULTY_NUM  = 18;
```
Unwanted side effect?

TrueBrain committed:
Codechange: merge guiflags and flags in settings .ini files

It was rather confusing which one was for what, especially as some
SaveLoad flags were settings-only. Clean up this mess a bit by
having only Setting flags.

TrueBrain committed:
Codechange: remove the unused SLF_HEX flag

frosch123 commented on src/saveload/saveload.h:
```diff
@@ -483,11 +483,9 @@ enum VarTypes {
-	SLF_NO_NETWORK_SYNC = 1 << 10, ///< do not synchronize over network (but it is saved if SLF_NOT_IN_SAVE is not set)
-	SLF_ALLOW_CONTROL   = 1 << 11, ///< allow control codes in the strings
-	SLF_ALLOW_NEWLINE   = 1 << 12, ///< allow new lines in the strings
+	SLF_NO_NETWORK_SYNC = 1 <<  8, ///< Do not synchronize over network.
```
There are now both SLF_ and SF_NO_NETWORK_SYNC :)

TrueBrain commented on src/saveload/saveload.cpp:
```diff
@@ -578,7 +578,7 @@ static inline uint SlGetArrayLength(size_t length)
  * @return Return the size of this type in bytes
  */
-static inline uint SlCalcConvMemLen(VarType conv)
+uint SlCalcConvMemLen(VarType conv)
```
In #9335 I revert this back to a `static inline`, but for my sanity (and yours) in this PR I first make it available for `settings.cpp`.

TrueBrain committed:
Codechange: move SLF_NOT_IN_SAVE into settings

It is a settings-only flag, so don't pollute SaveLoad code with it.

TrueBrain committed:
Codechange: move SLF_NO_NETWORK_SYNC into settings

It is a settings-only flag, so don't pollute SaveLoad code with it.
